### ISSUE TITLE
Centrifugo vs RabitMQ

### ISSUE THREAD

**Issue by chavenor** [centrifugal/centrifugo#126]
What are the major differences between Centrifugo and RabitMQ? 

I'm most interested in how it stacks up again RabitMQ in terms of client disconnects and the guarantee that the user received the message. 

https://www.rabbitmq.com/features.html

Thanks pretty neat package. 

Casey

**Comment by FZambia** [comment]
@chavenor hello!

Centrifugo and RabbitMQ are completely different purpose servers. Centrifugo's goal is delivering real-time messages to end users of your application with at most once delivery model (fire and forget in general). It has some mechanisms to recover missed messages after short network disconnects though but it's anyway a best-effort transport. Most of real-time client apps can tolerate this (of course there are exceptions). To ensure guaranteed message delivery to client when using Centrifugo additional work must be done on application backend side when needed. For example sending acks manually to your backend.

Centrifugo is just a PUB/SUB server designed for client applications, it's not suitable for usage on backend side - for example to coordinate microservices. Rabbit can be used as queue, PUB/SUB broker etc. Generally RabbitMQ is something that Centrifugo could be built on top of.

Centrifugo has some builtin features to help developing real-time apps - authorization, presence, join/leave events, browser/iOS/Android client library.

Maybe I have not answered on your question - then try to ask more concrete

**Comment by chavenor** [comment]
Got it.  My largest question was about the reliability but I can code that into my apps backed so we should be good to go.  I look forward to getting those performance gains.

**Comment by FZambia** [comment]
>  I look forward to getting those performance gains.

Sorry, don't understand  - what do you mean saying this?:)

**Comment by chavenor** [comment]
Sorry about that.  From a in/out messaging system I expect you to beat my current setups performance.

**Comment by FZambia** [comment]
Still don't understand your thoughts about performance - maybe English language barrier. Could you rephrase?

**Comment by chavenor** [comment]
The current system I'm using to handle real-time messages is likely much slower than Centrifugo.   Thus your (Centrifugo) should be much faster and overall perform better.

**Comment by FZambia** [comment]
Ah, comparing to another real-time messaging system. Now I see:) Hope it will really perform better for your case.

**Comment by chavenor** [comment]
Me too!  :)
